ts/WoltLabSuite/Core/Ui/Poll/Manager/Poll.ts

```diff
@@ -0,0 +1,205 @@
+    }
+
+    if (this.canViewParticipants()) {
+      this.participants = new Participants(this);
+    }
+  }
+
+  public getElement(): HTMLElement {
+    return this.poll;
+  }
+
+  public hasView(key: PollViews): boolean {
+    return this.views.has(key);
+  }
+
+  public getView(key: PollViews): HTMLElement {
+    if (!this.hasView(key)) {
+      throw new Error(`The view "${key}" is unknown for poll "${this.pollID}".`);
+    }
+
+    return this.views.get(key)!;
+  }
+
+  public displayView(key: PollViews): void {
+    if (!this.hasView(key)) {
+      throw new Error(`The view "${key}" is unknown for poll "${this.pollID}".`);
+    }
+
+    this.views.forEach((view) => {
+      view.hidden = true;
+    });
+
+    this.views.get(key)!.hidden = false;
+
+    this.voteView?.checkVisibility(key);
+    this.resultsView?.checkVisibility(key);
+    this.voteHandler?.checkVisibility(key);
+
+    if (!this.participants && this.canViewParticipants()) {
+      this.participants = new Participants(this);
+      this.participants.showButton();
+    }
+  }
+
+  public addView(key: PollViews, html: string): void {
+    const container = document.createElement("div");
+    container.dataset.key = key;
+    container.hidden = true;
+
+    DomUtil.setInnerHtml(container, html);
+
+    this.getInnerContainer().append(container);
+
+    if (this.views.has(key)) {
+      this.views.get(key)!.remove();
+    }
+
+    this.views.set(key, container);
+
+    if (key === PollViews.vote) {
+      this.voteHandler!.initSelects();
+    }
+  }
+
+  private canViewParticipants(): boolean {
+    return this.canViewResults && this.isPublic;
+  }
+
+  private getInnerContainer(): HTMLElement {
+    const innerContainer = this.poll.querySelector<HTMLElement>(".pollInnerContainer");
+
+    if (!innerContainer) {
+      throw new Error(`Could not find inner container for poll "${this.pollID}"`);
+    }
+
+    return innerContainer;
+  }
+
+  public changeTotalVotes(votes: number, tooltip: string): void {
+    const badge = this.getElement().querySelector<HTMLSpanElement>(".pollTotalVotesBadge");
+
+    if (!badge) {
+      throw new Error(`Could not find total votes badge.`);
+    }
+
+    badge.textContent = formatNumeric(votes);
+    badge.dataset.tooltip = tooltip;
+  }
+
+  get isPublic(): boolean {
+    return this.poll.dataset.isPublic === "true";
+  }
+
+  get maxVotes(): number {
+    return parseInt(this.poll.dataset.maxVotes!, 10);
+  }
+
+  get question(): string {
+    return this.poll.dataset.question!;
+  }
+
+  get canVote(): boolean {
+    return this.poll.dataset.canVote === "true";
+  }
+
+  set canVote(canVote: boolean) {
+    this.poll.dataset.canVote = canVote ? "true" : "false";
+  }
+
+  get canViewResults(): boolean {
+    return this.poll.dataset.canViewResult === "true";
+  }
+
+  set canViewResults(canViewResults: boolean) {
+    this.poll.dataset.canViewResult = canViewResults ? "true" : "false";
+  }
+}
+
+export class PollSetup {
```

Comment:
What is the intention behind this “one-off” class instead of using a plain function together with module level const map?